notifications/views/emailtemplate.py

```diff
@@ -259,3 +260,78 @@ def get(self, request, *args, **kwargs):
     def post(self, request, *args, **kwargs):
         view = ResendEmailTrigger.as_view()
         return view(request, *args, **kwargs)
+
+
+class ViewSentNotificationForCompany(NotificationsBaseView, generic.DetailView):
+    template_name = 'notifications/template/sent_notifications.html'
+    email_template_id_list = []
+
+    def get_object(self):
+        return get_object_or_404(CycleEmailTemplate,
+                                 id=self.kwargs['pk'])
+
+    def get_company(self):
+        return get_object_or_404(Company,
+                                 id=self.kwargs['pk_company'],
+                                 group=self.get_object().group)
+
+    def get_cycle_notification_template(self, stage, company, person):
+        email_template = EmailTemplate.objects.get(
+            group=company.group,
+            stage=Stage.objects.get(code=stage)
+        )
+
+        cycle_email_template = CycleEmailTemplate.objects.get(
+            emailtemplate=email_template,
+            cycle__year=timezone.now().year
+        )
+        self.email_template_id_list.append(cycle_email_template.id)
+
+        cycle_notification = CycleNotification.objects.filter(
+            emailtemplate=cycle_email_template,
+            email=person.email
+        )
+
+        return cycle_notification
+
+    def verify_cycle_notification(self, cycle_notification):
+        if cycle_notification.count() > 0:
+            return True
+        else:
+            return False
+
+    def get_persons_info(self, company, persons):
+        persons_info = []
+
+        for person in persons:
+            obj = dict()
+            obj['person'] = person
+            obj['stages'] = []
+
+            stages = Stage.objects.all()
+
+            # Skip first and last stage - meaning INITIATED and CLOSED
+            for i in range(1, 5):
```

Comment:
You have custom logic here that's hard to find - in case this rule changes/is reused. Move it to a method on Stage model.